src/Orleans.Transactions/State/TransactionQueue.cs

```diff
@@ -496,94 +498,100 @@ public int BatchableOperationsCount()
 
         private async Task StorageWork()
         {
-            try
+            // Stop if this activation is stopping/stopped.
+            if (this.activationLifetime.OnDeactivating.IsCancellationRequested) return;
+
+            using (this.activationLifetime.BlockDeactivation())
```

Comment:
Not much of a concern about this going to deactivated between?  Will BlockDeactivation() throw if the deactivation has started?

Reply:
It won't, it's just a hint to not deactivate while some block of code is executing. It's currently internal to tx asm